src/common/snippets/src/op/subgraph.cpp

```diff
@@ -92,6 +94,16 @@ auto snippets::op::Subgraph::wrap_node_as_subgraph(const std::shared_ptr<ov::Nod
 
     auto body_node = node->clone_with_new_inputs(body_inputs);
     body_node->set_friendly_name(node->get_friendly_name());
+    NGRAPH_SUPPRESS_DEPRECATED_START
+    for (size_t i = 0; i < node->get_output_size(); i++) {
+        const std::string new_name = ngraph::op::util::get_ie_output_name(node);
+        auto& out_tensor = body_node->get_output_tensor(i);
+        out_tensor.set_name(new_name);
+        if (!node->get_output_tensor(i).get_names().empty()) {
+            out_tensor.set_names(node->get_output_tensor(i).get_names());
+        }
+    }
+    NGRAPH_SUPPRESS_DEPRECATED_END
```

Comment:
As I can see we already have more or less the same logic inside `update_out_tensor_name`. Can we reuse it? Or maybe update `update_out_tensor_name` to handle internal body as well?

Reply:
As we discuss offline, I have added separate function for copying output tensor names
Also I have added functional tests to cover several results + output tensor names